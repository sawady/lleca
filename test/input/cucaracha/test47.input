
fun tabla(ancho : Int, alto : Int) {
    j := 0
    while j < alto {
      i := 0
      while i < ancho {
          putChar(120)
          i := i + 1
      }
      putChar(10)
      j := j + 1
    }
}

fun main() {
    tabla(10, 8)
}

