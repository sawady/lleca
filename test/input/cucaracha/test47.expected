Cons(
  Function(
    tabla,
    Cons(
      Parameter(
        ancho,
        Int
      ),
      Cons(
        Parameter(
          alto,
          Int
        ),
        Nil
      )
    ),
    Unit,
    Cons(
      StmtAssign(
        j,
        ExprConstNum(
          0
        )
      ),
      Cons(
        StmtWhile(
          ExprLt(
            ExprVar(
              j
            ),
            ExprVar(
              alto
            )
          ),
          Cons(
            StmtAssign(
              i,
              ExprConstNum(
                0
              )
            ),
            Cons(
              StmtWhile(
                ExprLt(
                  ExprVar(
                    i
                  ),
                  ExprVar(
                    ancho
                  )
                ),
                Cons(
                  StmtCall(
                    putChar,
                    Cons(
                      ExprConstNum(
                        120
                      ),
                      Nil
                    )
                  ),
                  Cons(
                    StmtAssign(
                      i,
                      ExprAdd(
                        ExprVar(
                          i
                        ),
                        ExprConstNum(
                          1
                        )
                      )
                    ),
                    Nil
                  )
                )
              ),
              Cons(
                StmtCall(
                  putChar,
                  Cons(
                    ExprConstNum(
                      10
                    ),
                    Nil
                  )
                ),
                Cons(
                  StmtAssign(
                    j,
                    ExprAdd(
                      ExprVar(
                        j
                      ),
                      ExprConstNum(
                        1
                      )
                    )
                  ),
                  Nil
                )
              )
            )
          )
        ),
        Nil
      )
    )
  ),
  Cons(
    Function(
      main,
      Nil,
      Unit,
      Cons(
        StmtCall(
          tabla,
          Cons(
            ExprConstNum(
              10
            ),
            Cons(
              ExprConstNum(
                8
              ),
              Nil
            )
          )
        ),
        Nil
      )
    ),
    Nil
  )
)
